test-data/unit/check-unions.test

```diff
@@ -229,10 +230,10 @@ T = TypeVar('T')
 S = TypeVar('S')
 def u(x: T, y: S) -> Union[S, T]: pass
 
-a = None # type: Any
+a: Any
 
-reveal_type(u(C(), None))  # N: Revealed type is "__main__.C"
-reveal_type(u(None, C()))  # N: Revealed type is "__main__.C"
+reveal_type(u(C(), None))  # N: Revealed type is "Union[None, __main__.C]"
```

Comment:
Would it be better to keep this test non-strict-optional? Seems like it specifically tests that None isn't kept as part of the union.